src/harvesters/SaddlePoolManager.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "BoringSolidity/interfaces/IERC20.sol";
+
+import "../external/saddle/ISwap.sol";
+
+contract SaddlePoolManager {
+    function addLiquidity(
+        address swap,
+        uint256 amountIn,
+        uint256 minToMint
+    ) public returns (uint256 LPTokens) {
+        ISwap hopSwap = ISwap(swap);
+        IERC20 token0 = IERC20(hopSwap.getToken(0));
```

Comment:
is the token always 0 and the hToken always 1?

I would probably add the index as a params, especially if we want to support the real saddle in the future